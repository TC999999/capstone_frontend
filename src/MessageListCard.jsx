import { Link } from "react-router-dom";

const MessageListCard = ({ message }) => {
  return (
    <div className="message-card">
      <h3>
        <Link
          to={`/messages/conversation/item/${message.itemID}/users/${message.to}/and/${message.from}`}
        >
          From: {message.from} about item {message.itemName}{" "}
        </Link>
      </h3>
      <p>
        <b>Sent At:</b>
        {message.createdAt}
      </p>
    </div>
  );
};

export default MessageListCard;
