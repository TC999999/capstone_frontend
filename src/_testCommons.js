const TEST_USER = {
  username: "u1",
  firstName: "fn1",
  lastName: "ln1",
  email: "testemail1@gmail.com",
  address: "100 address",
  zipCode: "11111",
  regionOrState: "state1",
  city: "city1",
  country: "country1",
  latitude: "10.99876",
  longitude: "-99.99981",
  isAdmin: false,
  isFlagged: false,
  items: [
    {
      itemID: 1,
      name: "i1",
      initialPrice: 100,
      imageURL: "test_url_1",
      condition: "great",
      description: "test item",
    },
    {
      itemID: 2,
      name: "i2",
      initialPrice: 100,
      imageURL: "test_url_2",
      condition: "great",
      description: "test item",
    },
  ],
  reviews: [],
  reports: [],
  pastPurchases: [
    {
      purchaseID: 1,
      itemName: "i4",
      sellerUser: "u3",
      imageURL: "test_url_4",
      price: 150,
      soldAt: expect.any(Date),
      fromFlaggedUser: false,
    },
  ],
};

const TEST_ITEM = {
  id: 1,
  name: "i1",
  imageURL: "test_url_1",
  initialPrice: 100,
  condition: "great",
  description: "test item",
  sellerUser: "u1",
  isSold: false,
  types: [{ id: 1, name: "electronics" }],
  location: {
    city: "city1",
    address: "100 address",
    latitude: "10.99876",
    longitude: "-99.99981",
    regionOrState: "state1",
    country: "country1",
    zipCode: "11111",
  },
};

const TEST_ITEMS = [
  {
    id: 3,
    name: "i3",
    imageURL: "test_url_3",
    initialPrice: 100,
    condition: "great",
    description: "test item",
    sellerUser: "u2",
    types: [
      { id: 1, name: "electronics" },
      { id: 3, name: "books" },
    ],
    location: {
      city: "city2",
      regionOrState: "state2",
      country: "country2",
      zipCode: "22222",
    },
  },
  {
    id: 2,
    name: "i2",
    imageURL: "test_url_2",
    initialPrice: 100,
    condition: "great",
    description: "test item",
    sellerUser: "u1",
    types: [
      { id: 1, name: "electronics" },
      { id: 2, name: "movies" },
    ],
    location: {
      city: "city1",
      regionOrState: "state1",
      country: "country1",
      zipCode: "11111",
    },
  },
  {
    id: 1,
    name: "i1",
    imageURL: "test_url_1",
    initialPrice: 100,
    condition: "great",
    description: "test item",
    sellerUser: "u1",
    types: [{ id: 1, name: "electronics" }],
    location: {
      city: "city1",
      regionOrState: "state1",
      country: "country1",
      zipCode: "11111",
    },
  },
];

const TEST_MESSAGE = {
  id: 3,
  from: "u2",
  to: "u1",
  itemID: 1,
  itemName: "i3",
  body: "test body",
  createdAt: "2024-07-09 14:23:43.469126-04",
};

const TEST_MESSAGE_2 = {
  id: 3,
  from_username: "u2",
  to_username: "u1",
  itemID: 1,
  item_name: "i3",
  body: "test body",
  sent_at: "2024-07-09 14:23:43.469126-04",
};

const TEST_MESSAGES = {
  conversation: [
    {
      id: 2,
      from_username: "u3",
      to_username: "u1",
      item_name: "i4",
      itemID: 4,
      body: "test body",
      sent_at: "2024-07-09 14:23:43.469126-04",
    },
    {
      id: 1,
      from_username: "u1",
      to_username: "u3",
      item_name: "i4",
      itemID: 4,
      body: "test body",
      sent_at: "2024-07-09 14:23:43.469126-04",
    },
  ],
};

const TEST_REVIEW = {
  id: 1,
  reviewed: "u3",
  reviewer: "u1",
  rating: 10,
  body: "test review",
  madeAt: "2024-07-09 14:23:43.469126-04",
};

const TEST_REPORT = {
  id: 1,
  reportedUser: "u2",
  reporterUser: "u1",
  body: "test body",
  madeAt: "2024-07-09 14:23:43.469126-04",
  isCleared: false,
};

const TEST_REPORTS = [
  {
    id: 1,
    reportedUser: "u2",
    reporterUser: "u1",
    body: "test body",
    madeAt: "2024-07-09 14:23:43.469126-04",
    isCleared: false,
  },
  {
    id: 2,
    reportedUser: "u1",
    reporterUser: "u2",
    body: "test body",
    madeAt: "2024-07-09 14:23:43.469126-04",
    isCleared: false,
  },
];

const TEST_PURCHASE = {
  purchaseID: 1,
  itemName: "i4",
  sellerUser: "u3",
  imageURL: "test_url_4",
  price: 150,
  soldAt: "2024-07-09 14:23:43.469126-04",
  fromFlaggedUser: false,
};

module.exports = {
  TEST_USER,
  TEST_ITEM,
  TEST_ITEMS,
  TEST_MESSAGE,
  TEST_MESSAGE_2,
  TEST_MESSAGES,
  TEST_REPORT,
  TEST_REPORTS,
  TEST_REVIEW,
  TEST_PURCHASE,
};
