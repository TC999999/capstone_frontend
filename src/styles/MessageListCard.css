.message-card {
  border: solid 2px black;
  padding: 2px;
  background-color: rgb(177, 170, 170);
  margin: 5px;
}

.message-card a {
  text-decoration-line: none;
  font-size: 15px;
  color: blue;
}

@media (min-width: 800px) {
  .message-card a {
    font-size: 20px;
  }
}

.message-card a:hover {
  text-decoration-line: underline;
  color: purple;
}

.message-card a:active {
  color: aquamarine;
}
