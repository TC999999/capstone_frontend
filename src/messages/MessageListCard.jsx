import { Link } from "react-router-dom";
import "../styles/MessageListCard.css";

const MessageListCard = ({ message }) => {
  return (
    <div className="message-card">
      <h3>
        <Link
          to={`/messages/conversation/item/${message.itemID}/users/${message.to}/and/${message.from}`}
        >
          From: {message.from} about item {message.itemName}{" "}
        </Link>
      </h3>
      <p className="message-sent">
        <small>
          <b>Sent At: </b>
          {message.createdAt}
        </small>
      </p>
    </div>
  );
};

export default MessageListCard;
