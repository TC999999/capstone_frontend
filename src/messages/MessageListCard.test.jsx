import { render } from "@testing-library/react";
import MessageListCard from "./MessageListCard.jsx";
import { TEST_MESSAGE } from "../_testCommons.js";
import { MemoryRouter } from "react-router-dom";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <MessageListCard message={TEST_MESSAGE} />
    </MemoryRouter>
  );
});

it("should match snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <MessageListCard message={TEST_MESSAGE} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("should have a message card class", function () {
  const { container } = render(
    <MemoryRouter>
      <MessageListCard message={TEST_MESSAGE} />
    </MemoryRouter>
  );
  const card = container.querySelector(".message-card");
  expect(card).toBeInTheDocument();
});

it("should have a message card class", function () {
  const { getByText } = render(
    <MemoryRouter>
      <MessageListCard message={TEST_MESSAGE} />
    </MemoryRouter>
  );
  const title = getByText("From: u2 about item i3");
  expect(title).toBeInTheDocument();
});
