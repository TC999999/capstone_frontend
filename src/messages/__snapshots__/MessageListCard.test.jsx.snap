// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="message-card"
  >
    <h3>
      <a
        href="/messages/conversation/item/1/users/u1/and/u2"
      >
        From: u2 about item i3 
      </a>
    </h3>
    <p
      class="message-sent"
    >
      <small>
        <b>
          Sent At: 
        </b>
        2024-07-09 14:23:43.469126-04
      </small>
    </p>
  </div>
</DocumentFragment>
`;
